crates/sui-types/src/gas.rs

```diff
@@ -9,16 +9,27 @@ use crate::{
     object::{Object, Owner},
 };
 use move_core_types::{
-    gas_schedule::{
-        AbstractMemorySize, GasAlgebra, GasCarrier, GasPrice, GasUnits, InternalGasUnits,
-    },
+    gas_algebra::{GasQuantity, InternalGas, InternalGasPerByte, NumBytes, UnitDiv},
     vm_status::StatusCode,
 };
-use move_vm_types::gas_schedule::{GasStatus, INITIAL_COST_SCHEDULE};
+use move_vm_test_utils::gas_schedule::{GasStatus, GasUnit, INITIAL_COST_SCHEDULE};
 use once_cell::sync::Lazy;
 use schemars::JsonSchema;
 use serde::{Deserialize, Serialize};
-use std::convert::TryFrom;
+use std::{
+    convert::TryFrom,
+    ops::{Add, Mul},
+};
+
+pub type GasUnits = GasQuantity<GasUnit>;
+pub enum GasPriceUnit {}
+pub enum SuiGasUnit {}
+
+// pub type StorageGasPricePerByte = GasQuantity<UnitDiv<GasPriceUnit, Byte>>;
+pub type ComputeGasPricePerUnit = GasQuantity<UnitDiv<GasUnit, GasUnit>>;
```

Comment:
What is `UnitDiv`?

Reply:
Okay so with Victor's changes, the algebra is very very strict. You cannot multiply two quantities X, Y to get Z unless Y is defined as Z/X.
This is all well and good when everyone conforms, but our approach in the code was very loose so I kinda played some tricks to get things working here. All this will be re-tooled as a fast follow in [this](https://github.com/MystenLabs/sui/issues/4172)